arrows/vtk/mesh_coloration.cxx

```diff
@@ -0,0 +1,577 @@
+/*ckwg +29
+ * Copyright 2016 by Kitware, SAS; Copyright 2017-2020 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name Kitware, Inc. nor the names of any contributors may be
+ *    used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "mesh_coloration.h"
+
+#include <kwiversys/SystemTools.hxx>
+#include "vtkKwiverCamera.h"
+
+// VTK includes
+#include "vtkCellData.h"
+#include "vtkDoubleArray.h"
+#include "vtkFloatArray.h"
+#include "vtkImageData.h"
+#include "vtkVector.h"
+#include "vtkNew.h"
+#include "vtkPointData.h"
+#include "vtkPointDataToCellData.h"
+#include "vtkPolyData.h"
+#include "vtkPolyDataMapper.h"
+#include "vtkRenderer.h"
+#include "vtkRendererCollection.h"
+#include "vtkRenderWindow.h"
+#include "vtkSequencePass.h"
+#include "vtkSmartPointer.h"
+#include "vtkWindowToImageFilter.h"
+#include "vtkXMLImageDataWriter.h"
+#include "vtkXMLPolyDataWriter.h"
+
+
+// Other includes
+#include <algorithm>
+#include <numeric>
+#include <sstream>
+#include <iomanip>
+
+
+namespace
+{
+static char const* const BLOCK_VR = "video_reader";
+static char const* const BLOCK_MR = "mask_reader";
+
+//----------------------------------------------------------------------------
+/// Compute median of a vector
+template <typename T>
+static void ComputeMedian(std::vector<T> vector, double& median)
+{
+  std::sort(vector.begin(), vector.end());
+  size_t middleIndex = vector.size() / 2;
+  if (vector.size() % 2 == 0)
+    {
```

Comment:
Indentation of the braces is non-standard here.  This is minor enough to fix in your follow-up PR.